eth2/lmd_ghost/src/lib.rs

```diff
@@ -52,4 +52,10 @@ pub trait LmdGhost<S: Store, E: EthSpec>: Send + Sync {
     /// Returns `Ok(())` if the underlying fork choice has maintained its integrity,
     /// `Err(description)` otherwise.
     fn verify_integrity(&self) -> Result<()>;
+
+    /// Encode the `LmdGhost` instance to bytes.
+    fn as_bytes(self) -> Vec<u8>;
```

Comment:
Given that this consumes `self`, we'll need to make `ReducedTree` clone so we can back it up without destroying it.